dask/tests/test_order.py

```diff
@@ -535,3 +541,43 @@ def test_map_overlap(abcde):
     o = order(dsk)
 
     assert o[(b, 1)] < o[(e, 5)] or o[(b, 5)] < o[(e, 1)]
+
+
+def test_use_structure_not_keys(abcde):
+    """ See https://github.com/dask/dask/issues/5584#issuecomment-554963958
```

Comment:
Can you add Closes https://github.com/dask/dask/issues/5584 to the original post, or is there remaining work to be done?